### PR TITLE
Allow PreferServerCipherSuites as a TLS Option

### LINKED ISSUE #6228: Cipher Suites Order missing
### Do you want to request a *feature* or report a *bug*?

Feature

### What did you expect to see?

I'm missing the option "PreferServerCipherSuites" to enable cipher suites order by server, like here https://forge.tedomum.net/tedomum/peertube/commit/1dd59831f80ff4d49f3b60c8c3b2aabfb1512eeb

Example:

```
[tls.options.extended]
    sniStrict = true
    minVersion = "TLSv1"
    PreferServerCipherSuites = true
    cipherSuites = [
        "TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA",
        "TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA",
        "TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256",
        "TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA384",
        "TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA",
        "TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA",
        "TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256",
        "TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA384",
        "TLS_DHE_RSA_WITH_AES_128_CBC_SHA",
        "TLS_DHE_RSA_WITH_AES_256_CBC_SHA",
        "TLS_DHE_RSA_WITH_AES_128_CBC_SHA256",
        "TLS_DHE_RSA_WITH_AES_256_CBC_SHA256",
        "TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256",
        "TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256",
        "TLS_DHE_RSA_WITH_AES_128_GCM_SHA256",
        "TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384",
        "TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384",
        "TLS_DHE_RSA_WITH_AES_256_GCM_SHA384"
    ]
```

As it is now, it's only enabled (and nowhere documented) when specifying the maxVersion parameter:

[https://github.com/containous/traefik/pull/5650/files/6f6760a529f8b2c04df0fed8475d7d6c0a8561a5](https://github.com/containous/traefik/pull/5650/files/6f6760a529f8b2c04df0fed8475d7d6c0a8561a5)
maxversion

### PR DESCRIPTION
**Pull request by dtomcej**
### What does this PR do?

This PR Allows the configuration of `PreferServerCipherSuites` independently of other TLS options.

### Motivation

Fixes #6228 

### More

- [ ] Added/updated tests
- [x] Added/updated documentation

### CODE REVIEW

**Comment by jbdoumenjou** [comment]
Could you update the reference/dynamic-configuration for both the file and KubernetesCRD with the new option?

File: pkg/provider/kubernetes/crd/traefik/v1alpha1/tlsoption.go [hunk 1]
```diff
@@ -19,12 +19,13 @@ type TLSOption struct {
...
-	CurvePreferences []string   `json:"curvePreferences,omitempty"`
-	ClientAuth       ClientAuth `json:"clientAuth,omitempty"`
-	SniStrict        bool       `json:"sniStrict,omitempty"`
+	MinVersion               string     `json:"minVersion,omitempty"`
+	MaxVersion               string     `json:"maxVersion,omitempty"`
+	CipherSuites             []string   `json:"cipherSuites,omitempty"`
+	CurvePreferences         []string   `json:"curvePreferences,omitempty"`
+	ClientAuth               ClientAuth `json:"clientAuth,omitempty"`
+	SniStrict                bool       `json:"sniStrict,omitempty"`
+	PreferServerCipherSuites bool       `json:"preferServerCipherSuites,omitempty"`
```
Comment (juliens) [inline pkg/provider/kubernetes/crd/traefik/v1alpha1/tlsoption.go]: You need to handle this in the "real" TLSOption creation 
https://github.com/containous/traefik/blob/master/pkg/provider/kubernetes/crd/kubernetes.go#L495

Comment (dtomcej) [reply pkg/provider/kubernetes/crd/traefik/v1alpha1/tlsoption.go]: Aye. Good catch.

Comment (dtomcej) [reply pkg/provider/kubernetes/crd/traefik/v1alpha1/tlsoption.go]: Added more tests to confirm 😄

**Pull request merged**